### PR TITLE
fix(gatsby-plugin-sharp): tracedSVG for `fluid` images should respect forced aspect ratio

### PR DESCRIPTION
**Pull request by fk**
Ref. #9204, not sure if this is the right way to go…

- fixes memoization
- fixes `examples/using-gatsby-image` CSS that allows seeing the tracedSVG on hover on https://using-gatsby-image.gatsbyjs.org/traced-svg/
- split the [traced-svg](https://using-gatsby-image.gatsbyjs.org/traced-svg/) gallery into two—cropped and uncropped—to demonstrate the fix

### CODE REVIEW

File: examples/using-gatsby-image/src/pages/traced-svg.js [hunk 1]
```diff
@@ -57,8 +57,9 @@ const UnsplashMasonry = edges => (
             "& .gatsby-image-wrapper:hover": {
               "& div + img": {
                 opacity: `1 !important`,
+                transition: `none !important`,
               },
-              "& img + img": {
+              "& img + picture > img": {
```
Comment (fk) [inline examples/using-gatsby-image/src/pages/traced-svg.js]: Above brings back this (right image is hovered): 
image

**Comment by nakedible** [comment]
Not necessarily to fix in this pull, but `healOptions` is kinda freaky:

- If the incoming options does not have `height` or `width`, then `width: 400` is added, even if the object has `maxHeight` and/or `maxWidth`.
- If the incoming options does not have `maxHeight` or `maxWidth`, then `maxWidth: 800` is added, even if the object has `height` and/or `width`.
- If the incoming options has `width` set as string, it is converted from string to number. Same for `height`. But if it has both, then *only* `width` is converted to number, but `height` remains as string.
- Same as above for `maxWidth` and `maxHeight`.

This pull sort of works around that by overwriting `width` and `height` from `maxWidth` and `maxHeight`, but still leaving in both. For example, if none of the width/height/maxWidth/maxHeight are specified, the resulting options object will have `width: 400` and `maxWidth: 800`.

**Comment by pieh** [comment]
@nakedible agree, it would be great to split healOptions because right now it tries to "heal" sanitize input for few different functions (eg. `fluid` which is where maxWidth/maxHeight is used and `fixed` or `resize` where width/height) is used, but it's not in scope of this PR.

**Comment by fk** [comment]
I can't exactly say why, but I'm hesitant to merge this…

**Comment by fk** [comment]
Hey @nakedible, I wanted to let you know that @pieh and me took another look at this PR last week and it seemed while this approach is working on 80% of the cropped images, it has some slight offset issues with the other 20%. :-/

The good news is that @pieh worked his magic and did a PoC for a proper fix that will hopefully replace this PR soon.
Please have a little more patience. 🙏

**Comment by wardpeet** [comment]
@pieh or @fk any progress? should we drop this Pr for now?

**Pull request merged**